central/role/datastore/datastore_impl.go

```diff
@@ -269,6 +276,14 @@ func (ds *dataStoreImpl) RemovePermissionSet(ctx context.Context, id string) err
 	return nil
 }
 
+func verifyPermissionSetOriginMatches(ps *storage.PermissionSet, requestOrigin storage.Traits_Origin) error {
+	if ps.GetTraits().GetOrigin() != requestOrigin {
+		return errors.Wrapf(errox.InvalidArgs, "permission set %q is %s, cannot be modified or deleted in %s manner",
```

Comment:
This sounds more sound to me?
```suggestion
		return errors.Wrapf(errox.InvalidArgs, "permission set %q's origin is %s and not %s, cannot be modified or deleted",
```